agent.go

```diff
@@ -715,6 +729,35 @@ func (ep *endpoint) deleteServiceInfoFromCluster(sb *sandbox, method string) err
 	return nil
 }
 
+func disableServiceInNetworkDB(a *agent, n *network, ep *endpoint) {
```

Comment:
disableServiceInNetworkDB - API handles many error conditions. But we are not propagating to the caller like deleteServiceInfoFromCluster. Dont we need to propagate any error messages to the caller ?

Reply:
Fair question.  In this case I think "no".  The main reason I put that logic in a separate function was for code style and to keep both the indent level and the length of deleteServiceInfoFromCluster() from growing too much.  However, some of the errors that said helper function can encounter are warning-level and some are error-level.  Returning an error back to the caller instead of doing the logging directly would require parsing the error type and issuing logs which would negate the readability benefits of using the helper function in the first place. 

If the result of any of these errors would have altered the control flow of deleteServiceInfoFromCluster(), then returning the error would have been absolutely the right thing to do.  However, there is really no way to recover from this error as evidenced by the handling in the `if fullremove {` branch above it.  (which was the state of the code before this patch).  If that branch encounters an error creating a cluster event it warns and moves on.

Also, for the record, the error-level logs in disableServiceInNetworkDB() are basically assertions.  The only reason they should ever trigger is that there is some critical error in our automatically-generated protocol buffer code.

Reply:
ok.  LGTM